osquery/tables/system/windows/shortcut_files.cpp

```diff
@@ -0,0 +1,214 @@
+      }));
+
+  boost::system::error_code ec;
+  for (const auto& lnk : paths) {
+    boost::filesystem::path path = lnk;
+    if (!boost::filesystem::is_regular_file(path, ec)) {
+      continue;
+    }
+    std::string lnk_content;
+    if (!readFile(path, lnk_content, 20).ok()) {
+      LOG(WARNING) << "Failed to read shortcut file: " << lnk;
+    }
+    std::stringstream check_ss;
+    for (const auto& hex_char : lnk_content) {
+      std::stringstream value;
+      value << std::hex << std::uppercase << (int)(unsigned char)(hex_char);
+      // Add additional 0 if single hex value is 0-F
+      if (value.str().size() == 1) {
+        check_ss << "0";
+      }
+      check_ss << value.str();
+    }
+    std::string header_sig = check_ss.str();
+
+    // Check for shortcut header size (0x0000004c) and GUID
+    // (00021401-0000-0000-c000-000000000046)
+    if (header_sig != "4C0000000114020000000000C000000000000046") {
+      continue;
+    }
+
+    // Read the whole shortcut file
+    lnk_content = "";
+    if (!readFile(path, lnk_content).ok()) {
+      LOG(WARNING) << "Failed to read shortcut file: " << lnk;
+    }
+    std::stringstream ss;
+    for (const auto& hex_char : lnk_content) {
+      std::stringstream value;
+      value << std::hex << std::uppercase << (int)(unsigned char)(hex_char);
+      // Add additional 0 if single hex value is 0-F
+      if (value.str().size() == 1) {
+        ss << "0";
+      }
+      ss << value.str();
+    }
+
+    const std::string lnk_hex = ss.str();
+    LinkFileHeader data;
+    TargetInfo target_data;
+    LocationInfo location_data;
+    ExtraDataTracker extra_data;
+    DataStringInfo data_info_string;
+
+    data = parseShortcutHeader(lnk_hex);
+    if (data.header == "") {
+      continue;
+    }
+    std::string data_string = lnk_hex.substr(152);
+    if (data.flags.has_target_id_list) {
+      target_data = parseTargetInfo(data_string);
+      data_string = target_data.data;
+    }
+    if (data.flags.has_link_info) {
+      location_data = parseLocationData(data_string);
+      data_string = location_data.data;
+    }
+    if (data.flags.has_name || data.flags.has_relative_path ||
+        data.flags.has_working_dir || data.flags.has_arguments ||
+        data.flags.has_icon_location) {
+      data_info_string = parseDataString(data_string,
+                                         data.flags.is_unicode,
+                                         data.flags.has_name,
+                                         data.flags.has_relative_path,
+                                         data.flags.has_working_dir,
+                                         data.flags.has_icon_location,
+                                         data.flags.has_arguments);
+      data_string = data_info_string.data;
+    }
+    extra_data = parseExtraDataTracker(data_string);
+
+    std::string target_path = "";
+    // Lookup and combine GUIDs and target file path
+    if (target_data.root_folder != "") {
```

Comment:
The nested if /else stuff is hard for me to follow

Reply:
I added a few additional comments, does that help a bit?
Basically if a GUID is found in the shellitem of the shortcut file, its doing a lookup for the proper GUID name and then merging it with the rest of the file path